docs/specifications/tests/RequirementsAndNormalizationOfDomainNames.md

```diff
@@ -0,0 +1,470 @@
+
+*Table 2: A summary of the valid ASCII character between labels using [Unicode]
+codes.*
+
+Unicode code or code range | Character or character range | Comment
+:--------------------------|:-----------------------------|:--------------------
+U+002E                     | .                            | [FULL STOP]
+
+The fact that "." (U+002E) character is the delimiter between labels puts some
+limitations on its use. The first label cannot be en empty label unless that is
+the only label, i.e. the domain name for the root name. With that exception (that
+is covered below) a domain name cannot have a "." initially. Only the last label
+can be an empty label (the root label), which means that there cannot be two or
+more consecutive "." in a valid domain name. The domain name, as entered to
+Zonemaster, can either have a final dot or not, and will normalized as described
+below.
+
+
+### IDN name
+
+A valid IDN name is a domain named where one or more labels are valid IDN label
+([RFC 5890][RFC 5890#2.3.2.3]) and the remaining labels are valid ASCII labels as
+defined above. An IDN label can be an A-label or a U-label
+([RFC 5890][RFC 5890#2.3.2.1], section 2.3.2.1).
+
+* A valid IDN name where all IDN labels are A-labels will automatically meet the
+ASCII name requirements above given that the non-IDN labels meet them.
+
+* A valid IDN name with one or more U-labels can be converted to a valid IDN name
+where all IDN labels are A-labels.
+
+A valid ASCII name is, by definition, encoded in ASCII. A valid IDN name must
+either be encoded in ASCII (no U-labels) or in UTF-8 (at least one U-label). If
+not, Zonemaster will not be able to process the domain name. Note that ASCII is a
+subset of UTF-8.
+
+A valid ASCII name consists, by definition, of only ASCII characters. A valid IDN
+name must either consists of only ASCII characters (no U-labels, only A-labels)
+or consist of at least one non-ASCII Unicode character in at least one label,
+i.e. at least one U-label. U-labels and A-labels can be mixed, and IDN labels can
+be mixed with non-IDN labels.
+
+
+### Length limitations
+
+There is a maximum length of the whole domain name and a maximum length of each
+label. These limitations are defined for a domain name of ASCII characters only,
+which means that any IDN U-label must be converted to the equivalent A-label
+before the limitations can be checked.
+
+The maximum total length of a domain name is 253 characters (or octets) if it
+has no final dot, 254 with the final dot ([RFC 1035][RFC 1035#2.3.4], section
+2.3.4). The RFC defines the limit as 255 octets, but that is the limitation in
+the DNS packet, where labels separation is done differently.
+
+The maximum length of a label is 63 characters (or octets),
+[RFC 1035][RFC 1035#2.3.4], section 2.3.4. A label must be at least one character
+(octet) long unless it is the label representing the root, which is zero in
```

Comment:
the root -> the root zone

Reply:
Not zone here, it is the name. I changed to "representing the root domain name".